mpibind.c

```diff
@@ -515,24 +509,120 @@ static char *get_gomp_str (hwloc_cpuset_t cpuset)
     return str;
 }
 
-static char *get_cuda_str (int32_t gpus, uint32_t gpu_bits)
+/*
+ * assign tasks to gpus and vice verso
+ * follows Edgar Leon Borja's algorithm from 'mpibind8'
+ */
+static char *get_gpustring( int32_t gpus, int32_t numas, uint32_t *numagroup )
 {
     char *str = NULL;
-    int32_t i, rc;
+    hwloc_obj_t obj;
+    int32_t mapped_numa=0, mapped_np_in_numa=0, mapped_id_in_numa=0;
+    uint32_t (*gpus_in_numa)[gpus] = malloc(sizeof(uint32_t[numas][gpus]));
+    uint32_t (*gpus_in_group)[gpus] = malloc(sizeof(uint32_t[numas][gpus]));
+    uint32_t *gpus_per_numa = calloc (numas, sizeof (uint32_t));
+    uint32_t *gpus_per_group = calloc (numas, sizeof (uint32_t));
```

Comment:
Maybe a leak here, are these allocations freed before exiting the function? Sorry if I just missed it.

Reply:
You didn't miss anything. I've just spent all of my time in scripting languages for too long. I'll free those and submit a new PR.

Reply:
'frees' committed. Guess I don't need to submit a new PR.